semihost/machine/hexagon/hexagon_exit.c

```diff
@@ -0,0 +1,17 @@
+#include "hexagon_semihost.h"
+#include <unistd.h>
+
+__noreturn void
+_exit(int status)
+{
+    /* simulator reads exit status from r2 */
+    /* simulator reads system call code from r0 */
+    register uintptr_t r0 __asm__("r0") = status;
+    __asm__ __volatile__("r2 = %0\n"
+                         "r0 = #24\n" SWI
+                         : "=r"(r0)
```

Comment:
Looks like `r0` is an input parameter, so it should be placed in the input section and use "r" (r0).